agent/service.go

```diff
@@ -154,7 +155,25 @@ func (ts *TaskService) Create(requestCtx context.Context, req *taskAPI.CreateTas
 		return nil, fmt.Errorf("Drive %q could not be found", driveID)
 	}
 
-	err = bundleDir.MountRootfs(drive.Path(), "ext4", nil)
+	const (
+		maxRetries = 3
+		retryDelay = 100 * time.Millisecond
```

Comment:
100ms seems like quite a long time to wait for the first retry. How did you arrive at this number?

Reply:
I was thinking originally 50ms, but decided on 100ms to be a little more safer. Are we happy with changing this to 50ms?

Reply:
I suspect even 25 is more than enough. I'd maybe start with 10.

Ideally we profile the `virtio_config_changed` kernel function in various scenarios (different numbers of device attachments, etc) and gather some specific numbers on which to base this decision...

Reply:
@nmeyerhans - I feel writing that benchmark is a pretty large task, no? Basically you'd need to write C to benchmark the kernel function, but you'd also have to make some sort of stub linux kernel. Am I thinking about this right? Or did you have something else in mind?

Reply:
I agree some actual metrics would be nice, but if we don't do the benchmarking right away, I'd ask we at least increase the total wait time to something more like 1 second (10 ms retry interval is good though). Just in an attempt to account for hosts that are slower and other unpredictable situations in which the VM takes more than 30ms to rescan the block device.

Reply:
Agreed; in the common case, the rescan operation finishes very quickly. We often don't need a retry at all. When we do need a retry, 10ms is likely to be more than sufficient, and gives us a positive user experience (the task starts quickly). In the case of a particularly slow host, a full second (or more) of maximum retries will ensure that we do eventually succeed. The only drawback there is that it takes a bit longer for fatal errors to become apparent, but I am OK with that.

Reply:
Fixed!

Reply:
JFTR, I've done a small amount of benchmarking of the `virtio_config_changed` kernel function. When patching a single drive, it completes in roughly 0.01ms. So... 10ms ought to be more than sufficient, even with lots of drives.